internal/k8s/reconciler/gateway.go

```diff
@@ -27,10 +22,11 @@ type K8sGateway struct {
 	*gwv1beta1.Gateway
 	GatewayState *state.GatewayState
 
-	logger   hclog.Logger
-	client   gatewayclient.Client
-	config   apigwv1alpha1.GatewayClassConfig
-	deployer *GatewayDeployer
+	logger    hclog.Logger
+	client    gatewayclient.Client
+	config    apigwv1alpha1.GatewayClassConfig
+	validator *validator.GatewayValidator
+	deployer  *GatewayDeployer
```

Comment:
Do you think you can add a `TODO` for removing these from the struct when we get to where this will be extracted into the manager? Just don't want to forget these. Same with the `client`/`logger` fields. Maybe we can just lump together the things to get dropped in subsequent PRs just so we know this is still in intermediate state.

Reply:
I can. Noting that the end result for part 2 of the refactor still has `deployer` living on this struct so I'm assuming it's removed in refactor part 3. Will likely wait to move the validator until the same time.

Reply:
Done